static/js/tools/timeline/page.tsx

```diff
@@ -15,250 +15,116 @@
  */
 
 import React, { Component } from "react";
-import { getPlaceNames, TimelineParams, ChartOptions } from "./util";
 import {
-  getStatsVarInfo,
-  getStatsVar,
-  StatsVarNode,
-  StatsVarInfo,
-} from "../statvar_menu/util";
+  getPlaceNames,
+  getTokensFromUrl,
+  addToken,
+  removeToken,
+  statVarSep,
+  placeSep,
+} from "./util";
+import { getStatsVarInfo, StatsVarInfo } from "../statvar_menu/util";
 import { SearchBar } from "./search";
-import { Menu } from "../statvar_menu/menu";
 import { Info } from "./info";
 import { ChartRegion } from "./chart_region";
-import {
-  NoopStatsVarFilter,
-  TimelineStatsVarFilter,
-  StatsVarFilterInterface,
-} from "../commons";
-import _ from "lodash";
+
+import { StatVarHierarchyType, NamedPlace } from "../../shared/types";
+import { StatVarHierarchy } from "../../browser/statvar_hierarchy";
 
 interface PageStateType {
-  statsVarNodes: StatsVarNode;
-  statsVarInfo: { [key: string]: StatsVarInfo };
-  placeIdNames: Record<string, string>; // [(placeId, placeName)]
-  statsVarTitle: Record<string, string>;
-  statsVarValid: Set<string>;
-  chartOptions: ChartOptions;
+  renderTrigger: number;
 }
 
-class Page extends Component<Record<string, unknown>, PageStateType> {
-  params: TimelineParams;
+class Page extends Component<unknown, PageStateType> {
+  placeName: Record<string, string>;
+  statVarInfo: Record<string, StatsVarInfo>;
 
-  constructor(props: Record<string, unknown>) {
+  constructor(props: unknown) {
     super(props);
-    this.handleHashChange = this.handleHashChange.bind(this);
-    this.params = new TimelineParams();
-    this.params.getParamsFromUrl();
-    // set default statsVarTitle as the statsVar dcids
-    const statsVarTitle = {};
-    for (const statsVar of this.params.getStatsVarDcids()) {
-      statsVarTitle[statsVar] = statsVar;
-    }
+    this.fetchDataAndRender = this.fetchDataAndRender.bind(this);
     this.state = {
-      placeIdNames: {},
-      statsVarValid: new Set(),
-      statsVarNodes: _.cloneDeep(this.params.statsVarNodes),
-      statsVarInfo: {},
-      statsVarTitle: statsVarTitle,
-      chartOptions: _.cloneDeep(this.params.chartOptions),
+      renderTrigger: 0,
```

Comment:
curious what this is being used for. It looks like it gets set and updated, but never really used?

Reply:
Use placeName and statVarIno directly in state. This was used in an initial version..